rasa/nlu/tokenizers/convert_tokenizer.py

```diff
@@ -30,21 +39,117 @@ class ConveRTTokenizer(WhitespaceTokenizer):
         "intent_split_symbol": "_",
         # Regular expression to detect tokens
         "token_pattern": None,
-        # Remote URL of hosted model
-        "model_url": TF_HUB_MODULE_URL,
+        # Remote URL/Local path to model files
+        "model_url": None,
     }
 
     def __init__(self, component_config: Dict[Text, Any] = None) -> None:
-        """Construct a new tokenizer using the WhitespaceTokenizer framework."""
+        """Construct a new tokenizer using the WhitespaceTokenizer framework.
 
+        Args:
+            component_config: User configuration for the component
+        """
         super().__init__(component_config)
 
-        self.model_url = self.component_config.get("model_url", TF_HUB_MODULE_URL)
+        self.model_url = self._get_validated_model_url()
 
         self.module = train_utils.load_tf_hub_model(self.model_url)
 
         self.tokenize_signature = self.module.signatures["tokenize"]
 
+    @staticmethod
+    def _validate_model_files_exist(model_directory: Text) -> None:
+        """Check if essential model files exist inside the model_directory.
+
+        Args:
+            model_directory: Directory to investigate
+        """
+        files_to_check = [
+            os.path.join(model_directory, "saved_model.pb"),
+            os.path.join(model_directory, "variables/variables.index"),
+            os.path.join(model_directory, "variables/variables.data-00001-of-00002"),
+            os.path.join(model_directory, "variables/variables.data-00000-of-00002"),
+        ]
+
+        for file_path in files_to_check:
+            if not os.path.exists(file_path):
+                raise RasaException(
+                    f"""File {file_path} does not exist.
+                    Re-check the files inside the directory {model_directory}.
+                    It should contain the following model 
+                    files - [{", ".join(files_to_check)}]"""
+                )
+
+    def _get_validated_model_url(self) -> Text:
```

Comment:
We could remove some indentation levels by doing something like this, what do you think?

```python
        if not model_url:
            raise RasaException(
                """Parameter "model_url" was not specified in the configuration of "ConveRTTokenizer".
                You can either use a community hosted URL of the model
                or if you have a local copy of the model, pass the
                path to the directory containing the model files."""
            )

        if model_url == ORIGINAL_TF_HUB_MODULE_URL:
            # Can't use the originally hosted URL
            raise RasaException(
                f"""Parameter "model_url" of "ConveRTTokenizer" was
                set to "{model_url}" which does not contain the model any longer.
                You can either use a community hosted URL or if you have a
                local copy of the model, pass the path to the directory
                containing the model files."""
            )

        if model_url == RESTRICTED_ACCESS_URL:
            # Can't use the URL that is reserved for tests only
            raise RasaException(
                f"""Parameter "model_url" of "ConveRTTokenizer" was
                set to "{model_url}" which is strictly reserved for pytests of Rasa Open Source only.
                Due to licensing issues you are not allowed to use the model from this URL.
                You can either use a community hosted URL or if you have a
                local copy of the model, pass the path to the directory
                containing the model files."""
                )

        if io.is_valid_remote_url(model_url):
            return model_url

        if os.path.isfile(model_url):
            # Definitely invalid since the specified path should be a directory
            raise RasaException(
                """Parameter "model_url" of "ConveRTTokenizer" was
                set to the path of a file which is invalid. You
                can either use a community hosted URL or if you have a
                local copy of the model, pass the path to the directory
                containing the model files."""
            )

        if os.path.isdir(model_url):
            # Looks like a local directory. Inspect the directory
            # to see if model files exist.
            self._validate_model_files_exist(model_url)
            # Convert the path to an absolute one since
            # TFHUB doesn't like relative paths
            return os.path.abspath(model_url)

        raise RasaException(
            f"""{model_url} is neither a valid remote URL nor a local directory.
            You can either use a community hosted URL or if you have a
            local copy of the model, pass the path to
            the directory containing the model files."""
        )
```